it/controllers/employment/EmployerPayAmountControllerISpec.scala

```diff
@@ -0,0 +1,341 @@
+      import user.specificExpectedResults._
+
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+
+        "should render How much did xxx pay you? page with empty error text when there no input" which {
+
+          implicit lazy val result: WSResponse = {
+            authoriseAgentOrIndividual(user.isAgent)
+            dropEmploymentDB()
+            insertCyaData(CyaModel.cya, User(mtditid, None, nino, sessionId, agentTest(user.isAgent)))
+            urlPost(urlEOY, welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)), body = Map[String, String]())
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an BAD_REQUEST status" in {
+            result.status shouldBe BAD_REQUEST
+          }
+
+          titleCheck(get.expectedErrorTitle)
+          inputFieldValueCheck("", inputAmountField)
+          errorSummaryCheck(get.emptyErrorText, expectedErrorHref)
+        }
+
+        "should render How much did xxx pay you? page with wrong format text when input is in incorrect format" which {
+
+          implicit lazy val result: WSResponse = {
+            authoriseAgentOrIndividual(user.isAgent)
+            dropEmploymentDB()
+            insertCyaData(CyaModel.cya, User(mtditid, None, nino, sessionId, "agent"))
+            urlPost(urlEOY, welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)), body = Map("amount" -> "|"))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an BAD_REQUEST status" in {
+            result.status shouldBe BAD_REQUEST
+          }
+
+          titleCheck(get.expectedErrorTitle)
+          inputFieldValueCheck("", inputAmountField)
+          errorSummaryCheck(wrongFormatErrorText, expectedErrorHref)
+        }
+
+        "should render How much did xxx pay you? page with max error ext when input > 100,000,000,000" which {
+
+          implicit lazy val result: WSResponse = {
+            authoriseAgentOrIndividual(user.isAgent)
+            dropEmploymentDB()
+            insertCyaData(CyaModel.cya, User(mtditid, None, nino, sessionId, "agent"))
+            urlPost(urlEOY, welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)),
+              body = Map("amount" -> "9999999999999999999999999999"))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an BAD_REQUEST status" in {
+            result.status shouldBe BAD_REQUEST
+          }
+
+          titleCheck(get.expectedErrorTitle)
+          inputFieldValueCheck("9999999999999999999999999999", inputAmountField)
+          errorSummaryCheck(maxAmountErrorText, expectedErrorHref)
+        }
+      }
+    }
+  }
+
+  ".submit" should {
+
+    "redirect to Overview page when a valid form is submitted" when {
```

Comment:
as above this test could be moved up to the prior .submit tests and utilise the us.isAgent to cover more scenarios when they're an agent